homeassistant/components/sensor/filter.py

```diff
@@ -0,0 +1,177 @@
+"""
+Support for filtering for sensor values.
+
+Example configuration:
+
+sensor:
+  - platform: simulated
+    name: 'simulated relative humidity'
+    unit: '%'
+    amplitude: 0 # Turns off sine wave
+    mean: 50
+    spread: 10
+    seed: 999
+
+  - platform: filter
+    entity_id: sensor.simulated_relative_humidity
+    name: lowpass
+    options:
+      time_constant: 10
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.filter/
+"""
+import asyncio
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_NAME, CONF_ENTITY_ID, ATTR_UNIT_OF_MEASUREMENT, STATE_UNKNOWN)
+from homeassistant.core import callback
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.filter import (
+    Filter, FILTERS)
+from homeassistant.helpers.event import async_track_state_change
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_PRE_FILTER = 'pre_filter_state'
+
+CONF_FILTER_OPTIONS = 'options'
+CONF_FILTER_NAME = 'filter'
+
+DEFAULT_NAME_TEMPLATE = "{} filter {}"
+ICON = 'mdi: chart-line-variant'
+
+# ALL filter arguments must be OPTIONAL
+FILTER_SCHEMA = vol.Schema({
+    vol.Optional('time_constant'): vol.Coerce(int),
```

Comment:
Validate periods of time with `cv.time_period`. Will still allow people to put in seconds but will also allow syntax like `1:23`. Returns a timedelta object, retrieve seconds with `value.total_seconds()`

Reply:
I have to disagree with this suggestion. 

The variable has time in its name, but no one uses time syntax to described it. Filters interface has been built to be as mathematical pure as possible, adding this option would make this design decision less soo :(